### ISSUE TITLE
RuntimeWarning: Parent module not found while handling absolute import

### ISSUE THREAD

**Issue by austinbyers** [airbnb/streamalert#267]
All of the StreamAlert Lambda functions output the following RuntimeWarning when running in AWS Lambda:

```
/var/task/stream_alert/alert_processor/main.py: RuntimeWarning: Parent module 'stream_alert.alert_processor' not found while handling absolute import
```

The warning happens because the full `stream_alert/[function]` folder structure in the repo does not exist when running in the Lambda environment. Each Lambda function sees only its own code, not the entire package structure. These errors pollute the logs, making to harder to find real errors.

One solution is to conditionally import based on whether the code is part of a package ([example](https://github.com/airbnb/binaryalert/blob/2d93266604140971df87e0e8ded71fb4be1ba095/lambda_functions/analyzer/main.py#L14-L20)). So Lambda imports like

```python
from stream_alert.alert_processor.helpers import validate_alert
```

could be replaced with
```python
if __package__:
    # Imported by other packages (e.g. unit tests)
    from stream_alert.alert_processor.helpers import validate_alert
else:
    # Running in AWS Lambda
    from helpers import validate_alert
```

**Comment by jacknagz** [comment]
> The warning happens because the full stream_alert/[function] folder structure in the repo does not exist when running in the Lambda environment.

However, the deployment package includes this structure:

![screen shot 2017-08-16 at 3 09 20 pm](https://user-images.githubusercontent.com/11466941/29387462-f5f9dbe8-8294-11e7-9ed7-21d44d804cc1.png)

**Comment by austinbyers** [comment]
Interesting. I wonder if Lambda's context is restricted to the folder containing the main entry point?

**Comment by jacknagz** [comment]
@austinbyers we might want to reach out to AWS on this

**Comment by strcrzy** [comment]
I'd like to know the answer to this too

**Comment by juhamust** [comment]
Any findings regarding to this?

**Comment by austinbyers** [comment]
Closed via #623
